src/pip/_internal/models/index.py

```diff
@@ -2,14 +2,22 @@
 
 
 class Index(object):
-    def __init__(self, url):
+    def __init__(self, url, file_storage_domains):
         self.url = url
         self.netloc = urllib_parse.urlsplit(url).netloc
         self.simple_url = self.url_to_path('simple')
         self.pypi_url = self.url_to_path('pypi')
 
+        # This is part of a temporary hack used to block installs of PyPI
+        # packages which depend on external urls only necessary until PyPI can
+        # block such packages themselves
+        self.file_storage_domains = file_storage_domains
+
     def url_to_path(self, path):
         return urllib_parse.urljoin(self.url, path)
 
 
-PyPI = Index('https://pypi.org/')
+PyPI = Index(
+    'https://pypi.org/',
+    ['files.pythonhosted.org', 'test-file.pythonhosted.org'],
```

Comment:
It's `test-files.pythonhosted.org`.

https://test-files.pythonhosted.org/packages/6d/00/8ed1b6ea43b10bfe28d08e6af29fd6aa5d8dab5e45ead9394a6268a2d2ec/requests-2.5.4.1-py2.py3-none-any.whl